app/jobs/foreman_statistics/trend_counter_job.rb

```diff
@@ -0,0 +1,21 @@
+module ForemanStatistics
+  class TrendCounterJob < ApplicationJob
+    def perform(options = {})
+      ForemanStatistics::TrendImporter.update!
+    ensure
+      self.class.set(wait: Setting[:outofsync_interval].minutes).perform_later(options)
```

Comment:
Should this use this setting? or perhaps create a new one for the plugin?

Reply:
Well in theory the best would be minimum from all of the sync intervals.
But that would be overcomplicating things I believe, so I used the default one.

Reply:
I would like to avoid having extra setting just for this. I don't see enough value in it.
I think even hardcoded `30.minutes` would be fine (that's what it was with the rake task).

Reply:
let's keep it hard coded then and add a setting only if needed.

Reply:
With a cronjob the user could edit the cron.d file, so it wasn't hardcoded. However, the `outofsync_interval` would be very surprising for me as a user. They aren't really related. You may not even use it and still want to get stats of the number of machines etc. I still don't see how outofsync_interval makes sense for something that doesn't run on a regular interval (like Ansible).

Btw, it looks like this implementation changes is that it's no longer a regular interval. If calculation takes a minute and it's 30 minutes, then the schedule is:
* 0:00
* 0:31
* 1:02
* 1:33
...

Reply:
True, we could start the timer at the beginning, but we do not want to do that not to overload the system :thinking:
In case it would take 31 minutes, the number of concurrent jobs would rise to infinity :thinking:

Reply:
I'm ambivalent on what's better in this case though.

Reply:
Perhaps you can take `max(interval - (now - start_time), 0)`. Maybe even better if you can specify that it should be executed after `start_time + interval`.

Reply:
Nice idea, I've implemented the `max` :+1: